src/utils/components/HardwareDevices/HardwareDevicesModal.tsx

```diff
@@ -46,100 +49,110 @@ const HardwareDevicesModal: React.FC<HardwareDevicesModalProps> = ({
-    const updatedVM = produceVMDevices(vm, (vmDraft: V1VirtualMachine) => {
-      if (type === HARDWARE_DEVICE_TYPE.GPUS) {
-        vmDraft.spec.template.spec.domain.devices.gpus = [...(devices || [])];
-      } else {
-        vmDraft.spec.template.spec.domain.devices.hostDevices = [...(devices || [])];
-      }
-    });
-    return updatedVM;
-  }, [devices, type, vm]);
+  const updatedVM = produceVMDevices(vm, (vmDraft: V1VirtualMachine) => {
+    vmDraft.spec.template.spec.domain.devices[type] = !isEmpty(devices) ? devices : null;
+  });
 
+  const hasNewDevices = devices?.some(
+    (device) => !isEmpty(device.name) || !isEmpty(device.deviceName),
+  );
+  const disableSubmit =
+    !isEmpty(devices) &&
+    devices?.some((device) => isEmpty(device.name) || isEmpty(device.deviceName));
   return (
     <TabModal
-      obj={updatedVirtualMachine}
+      obj={updatedVM}
       onSubmit={onSubmit}
       isOpen={isOpen}
       onClose={onClose}
-      headerText={showForm ? btnText : headerText}
+      headerText={headerText}
+      isDisabled={disableSubmit}
     >
+      {vmi && (
+        <ModalPendingChangesAlert
+          isChanged={getChangedDevices(updatedVM, vmi)?.length > 0 && hasNewDevices}
+        />
+      )}
       <Form>
-        {vmi && (
-          <ModalPendingChangesAlert
-            isChanged={getChangedDevices(updatedVirtualMachine, vmi)?.length > 0}
-          />
-        )}
-        {vm?.status?.created && vm?.spec?.running && (
-          <Alert
-            variant={AlertVariant.info}
-            isInline
-            title={t('Restart required to apply changes')}
-          >
-            {t(
-              'If you make changes to the following settings you will need to restart the virtual machine in order for them to be applied',
-            )}
-          </Alert>
-        )}
-        <HardwareDevicesList devices={devices} handleRemoveDevice={onRemoveDevice} />
-        {showForm && (
-          <Grid hasGutter>
-            <NameFormField name={name} setName={setName} />
-            <DeviceNameSelect
-              deviceName={deviceName}
-              setDeviceName={setDeviceName}
-              permittedHostDevices={permittedHostDevices}
-            />
-            <PlainIconButton
-              onClick={onConfirm}
-              icon={<CheckCircleIcon />}
-              fieldId="confirm-add-device"
-            />
-            <PlainIconButton onClick={onCancel} icon={<TrashIcon />} fieldId="cancel-add-device" />
+        <Trans t={t} ns="plugin__kubevirt-plugin">
+          <Text className="text-muted" component={TextVariants.p}>
+            Enter a name for the device to be assigned and select it from the dropdown menu. Click{' '}
+            <b> Save</b>.<br /> Click{' '}
+            <b> + Add {type === HARDWARE_DEVICE_TYPE.GPUS ? 'GPU' : 'Host'} device </b> to add
+            another devices.
+          </Text>
+        </Trans>
+        {devices.map(({ name, deviceName, deviceIndex }) => (
```

Comment:
mapping on the entire Grid causes rendering of more titles than needed, can you find a way to solve this?

Reply:
Thank you 👍 
I fixed this :)